lib/psci/psci_on.c

```diff
@@ -60,7 +61,7 @@ int psci_cpu_on_start(u_register_t target_cpu,
 			psci_plat_pm_ops->pwr_domain_on_finish);
 
 	/* Protect against multiple CPUs trying to turn ON the same target CPU */
-	psci_spin_lock_cpu(target_idx);
+	psci_lock_get(cpu_pd_node);
```

Comment:
The commit message misquotes the Arm ARM regarding synchronization primitives. The Arm ARM specifically talks about the LoadEx/StoreEx mechanism when the MMU is disabled, which is not the case here.

This patch also (unnecessarily as fas I can tell) adds locking to CPU nodes in addition to non-CPU-nodes, thereby blowing some more performance we've worked hard to achieve.

Reply:
I did point out that the quote was for the non MMU case, not necessarily this instance.

This patch does not add locking to CPU nodes, they where locking before, just using spinlocks unconditionally. All the other PSCI locking was done with either spinlocks or bakery locks depending on whether HW_ASSISTED_COHERENCY is set or not. All this patch does is cause this CPU lock to do the same, it will still use spinlocks when HW_ASSISTED_COHERENCY is set. (although other patches in this set do change that behavior as I believe that is not the right flag for this)

Reply:
> I did point out that the quote was for the non MMU case, not necessarily this instance.

You pointed it out in a PR comment but you haven't updated the commit message, and that misquote is the rational for this change.

> This patch does not add locking to CPU nodes...

The PSCI locks are meant to be used for state coordination, for which locking at the CPU level is not required. This patch add locks at that level (although they won't be used during state coordination). The other spinlocks (like here) are used for a different purpose so should be kept separate from the PSCI locks. As I said in another comment, spinlocks are architecturally guaranteed to work here, which is not the case for the PSCI locks on non-HW_ASSISTED_COHERENCY platforms.